maestro-app/src/main/java/bio/overture/maestro/app/infra/adapter/inbound/webapi/ManagementController.java

```diff
@@ -20,103 +20,134 @@
-import javax.inject.Inject;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-
 @Slf4j
 @RestController
 public class ManagementController {
 
-    private final Indexer indexer;
+  private final Indexer indexer;
 
-    @Inject
-    public ManagementController(Indexer indexer) {
-        this.indexer = indexer;
-    }
+  @Inject
+  public ManagementController(Indexer indexer) {
+    this.indexer = indexer;
+  }
 
-    @GetMapping
-    public Mono<Map<String, String>> ping() {
-        val response = new HashMap<String, String>();
-        response.put("status", "up");
-        return Mono.just(response);
-    }
+  @GetMapping
+  public Mono<Map<String, String>> ping() {
+    val response = new HashMap<String, String>();
+    response.put("status", "up");
+    return Mono.just(response);
+  }
 
-    @DeleteMapping("/index/repository/{repositoryCode}/study/{studyId}/analysis/{analysisId}")
-    @ResponseStatus(HttpStatus.OK)
-    public Mono<IndexResult> removeAnalysis(@PathVariable String analysisId,
-                                            @PathVariable String studyId,
-                                            @PathVariable String repositoryCode) {
-        log.debug("in removeAnalysis, args studyId {}, repoId: {}, analysisId : {}", studyId,
-            repositoryCode, analysisId);
-        return indexer.removeAnalysis(RemoveAnalysisCommand.builder()
+  @DeleteMapping("/index/repository/{repositoryCode}/study/{studyId}/analysis/{analysisId}")
+  @ResponseStatus(HttpStatus.OK)
+  public Mono<IndexResult> removeAnalysis(
+      @PathVariable String analysisId,
+      @PathVariable String studyId,
+      @PathVariable String repositoryCode) {
+    log.debug(
+        "in removeAnalysis, args studyId {}, repoId: {}, analysisId : {}",
+        studyId,
+        repositoryCode,
+        analysisId);
+    return indexer.removeAnalysis(
+        RemoveAnalysisCommand.builder()
             .analysisIdentifier(
                 AnalysisIdentifier.builder()
                     .repositoryCode(repositoryCode)
                     .analysisId(analysisId)
                     .studyId(studyId)
-                    .build()
-            ).build()
-        );
-    }
+                    .build())
+            .build());
+  }
 
-    @PostMapping("/index/repository/{repositoryCode}/study/{studyId}/analysis/{analysisId}")
-    @ResponseStatus(HttpStatus.CREATED)
-    public Mono<IndexResult> indexAnalysis(@PathVariable String analysisId,
-                                           @PathVariable String studyId,
-                                           @PathVariable String repositoryCode) {
-        log.debug("in indexAnalysis, args studyId {}, repoId: {}, analysisId : {}", studyId, repositoryCode, analysisId);
-        return indexer.indexAnalysis(IndexAnalysisCommand.builder()
+  @PostMapping(
+      "/index/file_centric/repository/{repositoryCode}/study/{studyId}/analysis/{analysisId}")
+  @ResponseStatus(HttpStatus.CREATED)
```

Comment:
So I actually got confused and made a mistake, I think we should not be asking for specific index update. but rather index on the enabled indexes .. so it makes sense to only have 1 index analysis http endpoint

Reply:
What if both indexes are enabled?